js/app/dashboard/redux/reducers.js

```diff
@@ -123,50 +125,53 @@ export const polylineEnabled = (state = {}, action) => {
 }
 
 export const selectedTasks = (state = [], action) => {
+  /*
+    FIXME selectedTasks is an array of task ids, not tasks objects
+  */
+  let newSelectedTasks
+
   switch (action.type) {
-  case TOGGLE_TASK:
+    case TOGGLE_TASK:
 
-    if (-1 !== state.indexOf(action.task['@id'])) {
-      if (!action.multiple) {
-        return []
+      if (-1 !== state.indexOf(action.task['@id'])) {
+        // let's remove this task!
+        if (!action.multiple) {
+          return []
+        }
+        return _.filter(state, task => task !== action.task['@id'])
       }
-      return _.filter(state, task => task !== action.task['@id'])
-    }
-
-    const newState = action.multiple ? state.slice(0) : []
-    newState.push(action.task['@id'])
-
-    return newState
 
+      newSelectedTasks = action.multiple ? action.currentlySelectedTasks.slice(0) : []
+      newSelectedTasks.push(action.task)
+      break
   case SELECT_TASK:
-
     if (-1 !== state.indexOf(action.task['@id'])) {
-
       return state
     }
 
-    return [ action.task['@id'] ]
-
+    return [action.task['@id']]
   case SELECT_TASKS:
-
-    return action.tasks.map(task => task['@id'])
-
-  case SELECT_TASKS_BY_IDS:
-
-    return action.taskIds
-
+    newSelectedTasks = action.tasks
```

Comment:
If possible, I suggest saving only task ids instead of the full objects to avoid duplication

Reply:
OK no problem

in this case we need to change `selectedTasks' state key to `selectedTasksIds`

Reply:
here I need tasks to perform validation on task data before changing the state (I dont return the tasks for state update)
maybe no the good place, let discuss it below